platform/kubernetes/kubernetes.go

```diff
@@ -79,43 +76,35 @@ func isAddon(obj namespacedLabeled) bool {
 
 // --- /add ons
 
+type Applier interface {
+	Delete(logger log.Logger, def []byte) error
+	Create(logger log.Logger, def []byte) error
+	Apply(logger log.Logger, def []byte) error
```

Comment:
The difference is that if you try to Create something that already exists, kubernetes will balk. So, if you calculate that you need to create something, but it breaks, your calculation was wrong and you should probably know about it.

A counter-argument is that actually, why should you care -- the goal is to make the cluster look like the config, and if applying something works whether or not it exists, just do that.

But the _counter_-counter argument is that applying something that doesn't exist and applying something that does do not _always_ have the same result. For example, with daemonsets, in the first case you get a daemonset configured how you expected; in the second case you get what you had already, which is different to the config (whereas using create would have told you something was wrong).